### PR TITLE
Find variants by trait

### PR DESCRIPTION
**Pull request by ussjohnjay**
1) This will get variants from traits_hidden instead of short_name, thus catching variants with no short_names or misleading short_names (e.g. Douglas Pabst, all Qs), and also crew with multiple possible variants (e.g. Duras Sisters, Data Q).

2) Variants will now be sorted by rarity and name.

3) This will also wrap variants to multiple lines for better responsive design on mobile (4 per row by default, down to 2 on smaller viewports).

This is how Data Q's variants currently look on mobile:

Screen Shot 2021-01-09 at 16 22 58

With this PR, the variants will wrap instead of overlapping (please ignore the broken images; I didn't change my env to show assets properly):

Screen Shot 2021-01-09 at 16 23 23

And the second variant group will now follow the first:

Screen Shot 2021-01-09 at 16 23 30

### CODE REVIEW

File: src/components/crewvariants.tsx [hunk 1]
```diff
@@ -17,33 +17,68 @@ class CrewVariants extends Component<CrewVariantsProps, CrewVariantsState> {
 	};
 
 	componentDidMount() {
+		// Get variant names from traits_hidden
+		let ignore = [
```
Comment (ineffyble) [inline src/components/crewvariants.tsx:21]: Do we have some mechanism for identifying new traits when they first appear so we can determine if they need to be added to this list?

Something to build into the "misc scripts" that handle crew updates, perhaps?

File: src/components/crewvariants.tsx [hunk 2]
```diff
@@ -17,33 +17,68 @@ class CrewVariants extends Component<CrewVariantsProps, CrewVariantsState> {
...
+						found.sort(function(a, b) {
+							if (a.max_rarity == b.max_rarity)
+								return a.name.localeCompare(b.name);
+							return a.max_rarity - b.max_rarity;
+						});
+						// short_name may not always be the best name to use, depending on the first variant
+						variants.push({'name': found[0].short_name, 'trait_variants': found});
+					}
+				});
+				self.setState({ 'variants': variants });
```
Comment (ineffyble) [inline src/components/crewvariants.tsx:54]: There's a magic shorthand for this 😃 

```suggestion
				self.setState({ variants });
```

**Comment by ineffyble** [comment]
Do you know if your formatting/layout improvements will solve this? 

image

File: src/components/crewvariants.tsx (hunk 2 above)
Comment (ussjohnjay) [reply src/components/crewvariants.tsx]: Nice, thanks!

**Comment by ussjohnjay** [comment]
Yes, I moved the link into a new block-level element.

File: src/components/crewvariants.tsx (hunk 1 above)
Comment (AlexCPU) [reply src/components/crewvariants.tsx:21]: @ineffyble & @ussjohnjay I've invited you both to the stt-datacore organisation so you should then be able to see the scripts repository.  I've had a little delve into it, and have no idea what's going on, or how it works.  If you can't work it out, you'll need to ask @TemporalAgent7 for help.

**Pull request merged**